### PR TITLE
Upgrade Azure.Identity to 1.10.3 and other dependencies

### PR DESCRIPTION
**Pull request by msJinLei**
## Description
Refer to PRs
https://github.com/Azure/azure-powershell/pull/23064
https://github.com/Azure/azure-powershell/pull/22669

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Authentication/Authentication/TokenCache/PowerShellTokenCacheProvider.cs [hunk 1]
```diff
@@ -174,7 +174,7 @@ public virtual IPublicClientApplication CreatePublicClient(string authority = nu
             if (AzureSession.Instance.TryGetComponent<IConfigManager>(nameof(IConfigManager), out var config)
                 && config.GetConfigValue<bool>(ConfigKeys.EnableLoginByWam))
             {
-                builder = builder.WithBrokerPreview();
+                builder = builder.WithBroker();
```
Comment (isra-fel) [inline src/Accounts/Authentication/Authentication/TokenCache/PowerShellTokenCacheProvider.cs:177]: Please check if we should use the override with parameter

File: src/Accounts/Authentication/Identity/AbstractAcquireTokenParameterBuilderExtensions.cs
Comment (isra-fel) [inline src/Accounts/Authentication/Identity/AbstractAcquireTokenParameterBuilderExtensions.cs:1]: Whenever you update Identiy source code, remember to update the README and describe the details

File: src/Accounts/Authentication/Properties/Resources.resx [hunk 2]
```diff
@@ -399,4 +399,7 @@
   <data name="HelpMessageOfCheckForUpgrade" xml:space="preserve">
     <value>When enabled, Azure PowerShell will check for updates automatically and display a hint message when an update is available. The default value is true.</value>
   </data>
+  <data name="FailedToMigrateMsalCacheWithLegayName" xml:space="preserve">
+    <value>INITIALIZATION: Failed to migrate MSAL token cache of the legacy name with error : {0}</value>
```
Comment (isra-fel) [inline src/Accounts/Authentication/Properties/Resources.resx:403]:
```suggestion
    <value>INITIALIZATION: Failed to migrate MSAL token cache of the legacy name with error: {0}. Run 'Connect-AzAccount' to sign in again.</value>
```

File: src/Accounts/Authentication/Identity/Validations.cs [hunk 3]
```diff
@@ -12,17 +12,16 @@
...
 
 namespace Microsoft.Azure.PowerShell.Authenticators.Identity
 {
     internal static class Validations
     {
-        private const string InvalidTenantIdErrorMessage = "Invalid tenant id provided. You can locate your tenant id by following the instructions listed here: https://learn.microsoft.com/partner-center/find-ids-and-domain-names";
+        private const string InvalidTenantIdErrorMessage = "Invalid tenant id provided. You can locate your tenant id by following the instructions listed here: https://docs.microsoft.com/partner-center/find-ids-and-domain-names";
 
-        private const string NullTenantIdErrorMessage = "Tenant id cannot be null. You can locate your tenant id by following the instructions listed here: https://learn.microsoft.com/partner-center/find-ids-and-domain-names";
+        private const string NullTenantIdErrorMessage = "Tenant id cannot be null. You can locate your tenant id by following the instructions listed here: https://docs.microsoft.com/partner-center/find-ids-and-domain-names";
```
Comment (wyunchi-ms) [inline src/Accounts/Authentication/Identity/Validations.cs:24]: The link should be `learn`

File: src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs [hunk 4]
```diff
@@ -43,14 +43,14 @@ public static void Initialize(string rootPath, IConditionalAssemblyContext conte
                 // todo: consider moving the list to a standalone config file
                 #region AssemblyList
                 CreateAssembly("netcoreapp2.1", "Azure.Core", "1.35.0.0").WithPowerShellCore(),
-                CreateAssembly("netcoreapp2.1", "Microsoft.Identity.Client", "4.49.1.0").WithPowerShellCore(),
-                CreateAssembly("netcoreapp3.1", "Microsoft.Identity.Client.Extensions.Msal", "2.23.0.0").WithPowerShellCore(),
```
Comment (isra-fel) [inline src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs:47]: We don't need to load different assemblies for different platforms anymore?

File: src/Accounts/Authentication/Authentication/TokenCache/PowerShellTokenCacheProvider.cs (hunk 1 above)
Comment (msJinLei) [reply src/Accounts/Authentication/Authentication/TokenCache/PowerShellTokenCacheProvider.cs:177]: The function is just to enable WAM. Another interface to customize WAM `public PublicClientApplicationBuilder WithBroker(BrokerOptions brokerOptions)` is only supported above NET6
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/25a286e3d963ccd913d67dcc0caa16047e9a65bb/src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs#L223

File: src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs (hunk 4 above)
Comment (msJinLei) [reply src/Accounts/AssemblyLoading/ConditionalAssemblyProvider.cs:47]: > We don't need to load different assemblies for different platforms anymore?

We use assembly for .net 2.0 as the replacement. Test it OK.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**